src/model.hpp

```diff
@@ -208,7 +208,8 @@ class model {
         current_spikes().get().push_back({source, tspike});
     }
 
-    void attach_sampler(cell_member_type probe_id, sampler_function f, time_type tfrom = 0) {
+    template <typename F>
+    void attach_sampler(cell_member_type probe_id, F&& f, time_type tfrom = 0) {
```

Comment:
If we're perfectly forwarding `F` to the (implicit) `std::function` constructor through the second argument of `cell_group::add_sampler`, I'd use `std::forward<F>(f)` below.

Otherwise, how about moving the definition of `sampler_function` outside of `cell_group` (but in the same header, or a common header) and use that?